document-portal/document-portal-fuse.c

```diff
@@ -3093,14 +3206,11 @@ xdp_fuse_init (GError **error)
 {
   /* Options:
    *  auto_unmount: Tell fusermount to auto unmount if we die.
-   *  splice_read: use splice() to read from fuse pipe
-   *  splice_write: use splice() to write to fuse pipe
-   *  splice_move: move buffers from writing app to kernel during splice write
-   *  atomic_o_trunc: We handle O_TRUNC in create()
-   *  big_writes: Allow > 4k writes
    */
-  char *fusermount_argv[] = { "xdp-fuse", "-osubtype=portal,fsname=portal,auto_unmount,splice_read,splice_write,splice_move,atomic_o_trunc,big_writes" };
+  char *fusermount_argv[] = { "xdp-fuse", "-osubtype=portal,fsname=portal,auto_unmount" };
   struct fuse_args args = FUSE_ARGS_INIT (G_N_ELEMENTS (fusermount_argv), fusermount_argv);
+  XdpFuseThreadData thread_data = {0};
+  struct fuse_cmdline_opts opts;
```

Comment:
I still think it would be good to move the args and opts variables over to xdp_fuse_thread rather than passing pointers to them through the XdpFuseThreadData struct.

If they exist on the fuse thread's stack, then there is absolutely no question that they'll be valid at the point where they are used. Putting their initialisation in one thread and use in the other just adds unnecessary complexity.

Reply:
Well one reason was to manage it's lifecycle there for both args and opts. Indeed can be split.
I might have also move args, but would need to define an auto-clear for that.